web/src/engine/websites/MangaLib.ts

```diff
@@ -1,46 +1,125 @@
+        url: string
+    }[]
+}
+
+type ChapterID = {
+    branch_id: string,
+    number: string,
+    volume: string
+}
+
+@Common.ImageAjax(true)
 export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = 'https://api.mangalib.me/api/';
+    private readonly siteId = 1;
+    private imageServer: ImageServer = undefined;
+    private readonly mangasTaskPool = new TaskPool(1, new RateLimit(2, 1));
 
     public constructor() {
-        super('mangalib', 'MangaLib', 'https://mangalib.org', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Russian, Tags.Source.Aggregator);
+        super('mangalib', 'MangaLib', 'https://mangalib.org', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Media.Manga, Tags.Language.Russian, Tags.Source.Aggregator);
     }
 
     public override get Icon() {
         return icon;
     }
 
+    public override async Initialize(): Promise<void> {
+        const { data: { imageServers } } = await FetchJSON<APIResult<ImageServers>>(new Request(new URL('constants?fields[]=imageServers', this.apiUrl)));
+        this.imageServer = imageServers.find(server => server.id === 'main' && server.site_ids.includes(this.siteId));
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/ru/manga/[^/]`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaSlug = new URL(url).pathname.split('/').at(-1);
+        const { data: { slug_url, rus_name } } = await FetchJSON<APIResult<APIManga>>(new Request(new URL(`manga/${mangaSlug}`, this.apiUrl)));
+        return new Manga(this, provider, slug_url, rus_name);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let page = 1, run = true; run; page++) {
+            const url = new URL(`manga?page=${page}&site_id[]=${this.siteId}`, this.apiUrl);
+            const { data, meta: { has_next_page } } = await this.mangasTaskPool.Add(() => FetchJSON<APIResult<APIManga[]>>(new Request(url)), Priority.Low);
+            mangaList.push(...data.map(manga => new Manga(this, provider, manga.slug_url, manga.rus_name.trim())));
+            run = has_next_page;
+        }
+        return mangaList;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const { data } = await FetchJSON<APIResult<APIChapter[]>>(new Request(new URL(`manga/${manga.Identifier}/chapters`, this.apiUrl)));
+        return data.reduce((accumulator: Chapter[], chapter) => {
+            let baseTitle = `Том ${chapter.volume} Глава ${chapter.number}`;
+            baseTitle += chapter.name ? ` - ${chapter.name}` : '';
+            const chapters = chapter.branches.map(branch => {
+                const teamName = chapter.branches_count > 1 ? `[${branch.teams.at(0).name}]` : '';
+                const chapterId = JSON.stringify({
```

Comment:
Suggestion to use URL path name including the corresponding search parameters, so it can be directly consumed in `FetchPages` instead of using intermediate JSON serialization.

Reply:
I can build a pathname if you want, but it wouldnt be the same one than the website. There is language code in pathname and i fear other Libgroup websites may use multiple language. Well, FetchPages dont need language 🤷